src/include/sof/debug/telemetry/telemetry.h

```diff
@@ -0,0 +1,86 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2024 Intel Corporation. All rights reserved.
+ */
+
+#ifndef __SOF_TELEMETRY_H__
+#define __SOF_TELEMETRY_H__
+
+/* Slot in memory window 2 to be used as telemetry slot */
+#define DW_TELEMETRY_SLOT 1
+/* Memory of average algorithm of performance queue */
+#define AVG_PERF_MEAS_DEPTH 64
+/* Number of runs taken to calculate average (algorithm resolution) */
+#define AVG_PERF_MEAS_PERIOD 16
+/* disables calculating systick_averages */
+#define PERFORMANCE_MEASUREMENTS
+
+struct system_tick_info {
+	uint32_t count;
+	uint32_t last_sys_tick_count;
+	uint32_t max_sys_tick_count;
+	uint32_t last_ccount;
+	uint32_t avg_utilization;
+	uint32_t peak_utilization;
+	uint32_t peak_utilization_4k;
+	uint32_t peak_utilization_8k;
+	uint32_t rsvd[2];
+};
+
+/*
+ * This is the structure of telemetry data in memory window.
+ * If you need to define a field, you should also define the fields before it to
+ * keep the internal structures aligned with each other.
+ */
+struct telemetry_wnd_data {
+	uint32_t separator_1;
+	struct system_tick_info system_tick_info[CONFIG_MAX_CORE_COUNT];
```

Comment:
How is this aligned with the coredump backend. See commit 0bff695d2c997363eb0c6cbf9421b056d645ea4a in zephyr by @pjdobrowolski . If we have users of the same slot both in Zephyr and SOF, that is a potential problem and conflicts are possible.

Reply:
Possible conflicts must be investigate. After  /* uint32_t separator_exception; */ we should have space where core dump should be placed without problems between zephyr and sof.

Reply:
Yes, coredump was here before we had telemetry structure, it's supposed to be placed at the end of it. Currently it just overwrites the structure, but only after the firmware crashes, so this memory is not really used anymore by then. Not much of an issue for now, it still works. We can realign coredump later.

Reply:
We can do it in this PR. It is connected by space in MW.

Reply:
@tobonex Can you confirm telemetry will not overwrite the coredump? This is minimum requirement to merge this PR.

Reply:
Currently it's the coredump that overwrites telemetry in case of a crash. Not great but functionally coredump should work the same. Still, my plan was to put coredump in its designed place in the window after this gets merged. Coredump is currently at the beginning of the window and apparently should be somewhere close to the end.

Reply:
Ack @tobonex . I just wanted to confirm it is tested that after coredump overwrites the telemetery, we won't have telemetry again overwrite the coredump. This would make coredump unsable in the build.

Reply:
I'll try to crash it a few times to check if it ever happens.